src/Commands/Site/ListCommand.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Site;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Terminus\Collections\Sites;
+
+class ListCommand extends TerminusCommand
+{
+    /**
+     * Looks up a site name
+     *
+     * @command site:list
+     * @alias sites
+     *
+     * @field-labels
+     *   name: Name
+     *   id: ID
+     *   service_level: Service Level
+     *   framework: Framework
+     *   owner: Owner
+     *   created: Created
+     *   memberships: Memberships
+     * @option team Filter for sites you are a team member of
+     * @option owner Filter for sites a specific user owns. Use "me" for your own user
+     * @option org Filter sites you can access via the organization. Use "all" to get all.
+     * @option name Filter sites you can acces via name.
+     * @usage terminus site:list
+     *   * Responds with list of every site you can access
+     *   * Responds with "You have no sites." if you have no sites
+     * @usage terminus site:list --team
+     *   * Responds with a list of sites you are a team member of
+     *   * Responds with a notice stating no sites match criteria if none exist
+     * @usage terminus site:list --owner=<uuid>
+     *   * Responds with a list of sites owned by the user with the given UUID
+     *   * Responds with a notice stating no sites match criteria if none exist
+     * @usage terminus site:list --owner=me
+     *   * Responds with a list of sites the logged-in user owns
+     *   * Responds with a notice stating no sites match criteria if none exist
+     * @usage terminus site:list --org=<id|name>
+     *   * Responds with a list of sites inside the organization(s) indicated
+     *   * Responds with a notice stating no sites match criteria if none exist
+     * @usage terminus site:list --org=all
+     *   * Responds with a list of sites belonging to organization you are a member of
+     *   * Responds with a notice stating no sites match criteria if none exist
+     * @usage terminus site:list --name=<regex>
+     *   * Responds with a list of sites you have access to by name
+     *   * Responds with a notice stating no sites match criteria if none exist
```

Comment:
@TeslaDethray should decide what the best interim approach here should be; as long as we are nominally consistent between commands, then it won't be hard to adapt when this text is inserted into the help output.